frontend/components/commons/dataset-deletion/DatasetDelete.component.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="description">
+    <h2 class="--heading5 --semibold description__title">{{ sectionTitle }}</h2>
+    <base-card
+      card-type="danger"
+      :title="datasetDeleteTitle"
+      text="Be careful, this action is not reversible"
+      buttonText="Delete dataset"
+      @card-action="showDeleteModal(true)"
+    />
+    <base-modal
+      class="delete-modal"
+      :modal-custom="true"
+      :prevent-body-scroll="true"
+      modal-class="modal-secondary"
+      :modal-title="modalTitle"
+      :modal-visible="isDeleteModalVisible"
+      @close-modal="showDeleteModal(false)"
+    >
+      <div>
+        <p v-html="modalDescription"></p>
+        <div class="modal-buttons">
+          <base-button class="primary outline" @click="showDeleteModal(false)">
+            Cancel
+          </base-button>
+          <base-button class="primary" @click="onConfirmDeleteDataset">
+            Yes, delete
+          </base-button>
+        </div>
+      </div>
+    </base-modal>
+  </div>
+</template>
+
+<script>
+import { mapActions } from "vuex";
+import { currentWorkspace } from "@/models/Workspace";
+import { getDatasetFromORM } from "@/models/dataset.utilities";
+export default {
+  props: {
+    datasetId: {
+      type: Array,
+      required: true,
+    },
+    datasetTask: {
+      type: String,
+      required: true,
+    },
```

Comment:
it's better to put only datasetId as props and remove datasetTask.
DatasetTaks can be computed from vuexorm (see how I did in edition label components)